### PR TITLE
WCMS-476 Made it so that Events don't need an end date

### PR DESCRIPTION
**Pull request by rift137**

### CODE REVIEW

File: spec/lib/buweb/event_spec.rb [hunk 1]
```diff
@@ -70,6 +69,27 @@ def event_in_index?(p)
...
     end
+
+    describe '#ends_after_it_starts' do
+      context 'event start date with no end date' do
+        let(:attrs) { {start_date: Time.now, end_date: nil} }
+        it { should be_valid }
+      end
+    end
+
+    describe '#ends_after_it_starts' do
```
Comment (halloffame) [inline spec/lib/buweb/event_spec.rb]: These should all be nested under the same describe block...

```ruby
describe '#ends_after_it_starts' do
  ...
end
```

File: lib/buweb/event.rb [hunk 2]
```diff
@@ -147,6 +148,10 @@ def site_categories_type
     end
   end
 
+  def ends_after_it_starts
+    errors.add :base, 'The event end date should be after the start date.' if end_date.present? && (end_date < start_date)
```
Comment (halloffame) [inline lib/buweb/event.rb]: I think you can add the error directly to the `:end_date` attribute

```ruby
errors.add :end_date, 'should be after the start date' if ...
```
Try that and see if it works. This is how the form knows which fields to add the error class to.

File: lib/buweb/event.rb [hunk 3]
```diff
@@ -78,15 +79,15 @@ def set_slug
   end
 
   def day_span
-    (end_date.to_date - start_date.to_date + 1).to_i
+    (end_date.to_date - start_date.to_date + 1).to_i if end_date.present?
   end
 
   def single_day_event?
-    day_span == 1
+    day_span == 1 if day_span
```
Comment (halloffame) [inline lib/buweb/event.rb]: Wouldn't an event with a start date and no end date be considered a single day event?

Comment (rift137) [inline lib/buweb/event.rb]: I would assume that that would be true but if an event extends throughout the night then it would be multiple days.

**Comment by rift137** [comment]
:+1:

**Pull request merged**